src/librustc_codegen_llvm/llvm/ffi.rs

```diff
@@ -774,7 +774,8 @@ extern "C" {
     pub fn LLVMIsAGlobalVariable(GlobalVar: &Value) -> Option<&Value>;
     pub fn LLVMAddGlobal(M: &'a Module, Ty: &'a Type, Name: *const c_char) -> &'a Value;
     pub fn LLVMGetNamedGlobal(M: &Module, Name: *const c_char) -> Option<&Value>;
-    pub fn LLVMRustGetOrInsertGlobal(M: &'a Module, Name: *const c_char, T: &'a Type) -> &'a Value;
+    pub fn LLVMRustGetOrInsertGlobal(M: &'a Module, Name: *const c_char, NameLen: size_t,
+                                     T: &'a Type) -> &'a Value;
```

Comment:
FWIW, I think there's an opportunity to convert a lot more of these `LLVMRust` wrappers to pointer+len strings, using `StringRef` on the actual C++ LLVM call, but for now I only changed a couple that were involved with the `ValueName` calls.

Reply:
Agreed. I also wouldn't object if you did the rest too but it's not necessary.

Reply:
I'll probably work on those incrementally, in separate PRs.